### PR TITLE
DiffInclude annotation

### PR DESCRIPTION
**Pull request by stefanicai**
@bartoszwalacik here's a draft, to get an idea where I'm going with this. I can finish it once you answer the question on Gitter.

### CODE REVIEW

**Comment by bartoszwalacik** [comment]
ok, I will take a look

File: javers-core/src/main/java/org/javers/core/metamodel/annotation/DiffInclude.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * class B extends A {
+ *     &#64;DiffIgnore
+ *     private String qux;
+ * }
+ *
+ * @see DiffIgnore
+ * @author Iulian Stefanica
+ */
+@Target({ TYPE})
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/core/metamodel/annotation/DiffInclude.java]: `@Target({METHOD, FIELD})` ?

File: javers-core/src/main/java/org/javers/core/metamodel/type/TypeFactory.java [hunk 2]
```diff
@@ -42,23 +43,19 @@ JaversType create(ClientsClassDefinition def) {
     JaversType create(ClientsClassDefinition def, ClassScan scan) {
         if (def instanceof CustomDefinition) {
             return new CustomType(def.getBaseJavaClass());
-        } else
-        if (def instanceof EntityDefinition) {
+        } else if (def instanceof EntityDefinition) {
             return entityTypeFactory.createEntity((EntityDefinition) def, scan);
-        } else
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/core/metamodel/type/TypeFactory.java]: could you please redo the reformatting? it makes code review harder

**Comment by bartoszwalacik** [comment]
I think we have it, what do you think, is it ok?

**Comment by stefanicai** [comment]
Think so, just wanted to add another test, but I'm travelling tomorrow and Tuesday.

File: javers-core/src/main/java/org/javers/core/metamodel/type/TypeFactory.java (hunk 2 above)
Comment (stefanicai) [reply javers-core/src/main/java/org/javers/core/metamodel/type/TypeFactory.java]: Do you have a formatting file that I can import in IntelliJ so I format code the same as you do?

**Comment by bartoszwalacik** [comment]
we need one more test for this case:

If some properties in a subclass should be included, apply the {@code DiffInclude} annotation on them.

**Pull request merged**